spec/integration/calls_api_integration_spec.rb

```diff
@@ -1,42 +1,180 @@
+      expect(response[DATA].disconnect_url).to eq(BASE_CALLBACK_URL + "/callbacks/disconnect")
+
+      $call_info_id = response[DATA].call_id
+      $active_calls.append($call_info_id)
     end
   end
 
   # Get Call State Information
-  describe 'get_call_state test' do
-    it 'should work' do
-      # assertion here. ref: https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers
+  describe 'get_call_state' do
+    it 'gets the call state' do
+      response = @api_instance_calls.get_call_state_with_http_info(BW_ACCOUNT_ID, $call_info_id)
+
+      expect(response[CODE]).to eq(200)
+      expect(response[DATA]).to be_instance_of(Bandwidth::CallState)
+      expect(response[DATA].call_id).to eq($call_info_id)
+      expect(response[DATA].account_id).to eq(BW_ACCOUNT_ID)
+      expect(response[DATA].application_id).to eq(BW_VOICE_APPLICATION_ID)
+      expect(response[DATA].start_time).to be_instance_of(Time)
+      expect(response[DATA].last_update).to be_instance_of(Time)
+      expect(response[DATA].state).to be_instance_of(String)
+      expect(response[DATA].direction).to eq(Bandwidth::CallDirectionEnum::OUTBOUND)
     end
   end
 
   # Update Call
-  describe 'update_call test' do
-    it 'should work' do
-      # assertion here. ref: https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers
+  describe 'update_call' do
+    it 'creates and updates a call' do    
+      update_call_body = Bandwidth::UpdateCall.new(
+        state: Bandwidth::CallStateEnum::ACTIVE,
+        redirect_url: MANTECA_BASE_URL + "/bxml/pause"
+      )
+      
+      update_call_id = create_manteca_call(@api_instance_calls)
+      sleep(SLEEP_TIME_S)
+      
+      update_response = @api_instance_calls.update_call_with_http_info(BW_ACCOUNT_ID, update_call_id, update_call_body)
+      expect(update_response[CODE]).to eq(200)
+      sleep(SLEEP_TIME_S)
+      
+      complete_response = @api_instance_calls.update_call_with_http_info(BW_ACCOUNT_ID, update_call_id, $complete_call_body)
+      expect(complete_response[CODE]).to eq(200)
     end
   end
 
   # Update Call BXML
-  describe 'update_call_bxml test' do
-    it 'should work' do
-      # assertion here. ref: https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers
+  describe 'update_call_bxml' do
+    it 'updates a call using bxml' do
+      update_call_id = create_manteca_call(@api_instance_calls)
+      sleep(SLEEP_TIME_S)
+
+      update_bxml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Bxml><SpeakSentence locale=\"en_US\" gender=\"female\" voice=\"susan\">This is a test bxml response</SpeakSentence><Pause duration=\"3\"/></Bxml>";
+      update_response = @api_instance_calls.update_call_bxml_with_http_info(BW_ACCOUNT_ID, update_call_id, update_bxml)
+      expect(update_response[CODE]).to eq(204)
+      sleep(SLEEP_TIME_S)
+      
+      complete_response = @api_instance_calls.update_call_with_http_info(BW_ACCOUNT_ID, update_call_id, $complete_call_body)
```

Comment:
Even though we are updating the call to be complete, should we still add this call id to our list and verify that it is cleaned up later?

Reply:
All of the calls that get created using the `create_manteca_call` get added to the list of active calls within that function. [Here](https://github.com/Bandwidth/ruby-sdk/pull/71/files#diff-4bab836705b9ad756b0d0299b05d329d01b52b7bdab59781c13fac9af9eca1c7R31) is where I do it.